thisisshub/rpmlint
# Add test in test_speccheck, modified SpecCheck.py

This commit contains,
1. Refactored check in SpecCheck.py

2. Source-rpms in test/source

3. Tests in test_speccheck which are named with test_check_* and 
    their subsequent test_check_*_not_found/applied listed as,
- test_check_no_spec_file
- test_check_no_spec_file_not_applied
- test_check_non_utf8_spec_file
- test_check_non_utf8_spec_file_not_applied
- test_check_invalid_spec_name
- test_check_invalid_spec_name_not_applied
- test_check_non_break_space
- test_check_non_break_space_not_applied
- test_check_rpm_buildroot_usage_under_prep
- test_check_rpm_buildroot_usage_under_build
- test_check_rpm_buildroot_usage_not_applied
- test_check_make_check_outside_check_section
- test_check_make_check_inside_check_section
- test_check_make_check_outside_not_applied
- test_check_setup_not_quiet
- test_check_setup_is_quiet
- test_check_setup_not_quiet_not_applied
- test_check_setup_not_in_prep
- test_check_setup_inside_prep
- test_check_setup_not_in_prep_not_applied
- test_check_autopatch_not_in_prep
- test_check_autopatch_in_prep
- test_check_autopatch_not_in_prep_not_applied
- test_check_autosetup_not_in_prep
- test_check_autosetup_inside_prep
- test_check_autosetup_not_in_prep_not_applied
- test_check_use_of_rpm_source_dir
- test_check_use_of_rsd_not_applied
- test_check_configure_without_libdir_spec
- test_check_cwls_not_applied
- test_check_hardcoded_library_path
- test_check_hclp_not_applied
- test_check_obsolete_tag
- test_check_obsolete_tag_not_applied
- test_check_hardcoded_path_in_buildroot_tag
- test_check_hpbt_not_applied
- test_check_buildarch_instead_of_exclusivearch_tag
- test_check_biet_not_applied
- test_check_hardcoded_packager_tag
- test_check_hardcoded_packager_tag_not_applied
- test_check_hardcoded_prefix_tag
- test_check_hardcoded_prefix_tag_not_applied
- test_check_buildprereq_use
- test_check_buildprereq_use_not_applied
- test_check_bsisr
- test_check_bsisr_not_applied
- test_check_forbidden_controlchar_found
- test_check_forbidden_controlchar_not_found
- test_check_coid
- test_check_coid_found
- test_check_coid_is_found
- test_check_coid_is_found_with_single_space
- test_check_coid_is_found_with_double_space
- test_check_unversioned_explicit_version
- test_check_unversioned_explicit_version_not_found
- test_check_uev_not_applied
- test_check_unversioned_explicit_obsoletes
- test_check_unversioned_explicit_obsoletes_not_found
- test_check_ueo_not_found
- test_check_macro_in_changelog
- test_check_macro_in_changelog_not_found
- test_check_macro_in_changelog_not_applied
- test_check_libdir_macro_in_noarch_package
- test_check_lmnp_not_applied
- test_check_deprecated_grep
- test_check_deprecated_grep_not_found
- test_check_deprecated_grep_not_applied
- test_check_macro_in_comment
- test_check_macro_in_comment_not_found
- test_check_macro_in_comment_not_applied
- test_check_no_build_root_tag
- test_check_no_build_root_tag_not_applied
- test_check_no_essential_section
- test_check_no_essential_section_not_applied
- test_check_more_than_one_changelog_section
- test_check_mtocs_not_applied
- test_check_lib_package_without_mklibname
- test_check_lpwm_not_applied
- test_check_depscript_without_disabling_depgen
- test_check_depscript_without_disabling_depgen_not_found
- test_check_dwdd_not_applied
- test_check_mixed_use_of_spaces_and_tabs
- test_check_mixed_use_of_spaces_and_tabs_not_applied
- test_check_ifarch_applied_patch
- test_check_ifarch_applied_not_enforced
- test_check_patch_not_applied
- test_check_patch_not_found
- test_check_patch_not_applied_not_enforced
- test_check_invalid_url
- test_check_invalid_url_not_applied

4. Specfile in test/spec listed as,
- %autopatch-not-in-prep.spec
- %autosetup-not-in-prep.spec
- broken-syntax-in-scriptlet-requires.spec
- buildarch-instead-of-exclusivearch-tag.spec
- buildprereq-use.spec
- comparison-operator-in-deptoken.spec
- configure-without-libdir-spec.spec
- deprecated-grep.spec
- depscript-without-disabling-depgen.spec
- forbidden-controlchar-found.spec
- hardcoded-library-path.spec
- hardcoded-packager-tag.spec
- hardcoded-path-in-buildroot-tag.spec
- hardcoded-prefix-tag.spec
- %ifarch-applied-patch.spec
- invalid-url.spec
- libdir-macro-in-noarch-package.spec
- lib-package-without-%mklibname.spec
- macro-in-changelog.spec
- macro-in-comment.spec
- make-check-outside-check-section.spec
- mixed-use-of-spaces-and-tabs.spec
- more-than-one-%changelog.spec
- no-buildroot-tag.spec
- non-break-space.spec
- non-standard-group.spec
- non-utf8-spec-file.spec
- no-%%%s-section.spec
- obsolete-tag.spec
- rpm-buildroot-usage.spec
- setup-not-in-prep.spec
- setup-not-quiet.spec
- SpecCheck2.spec
- SpecCheck3.spec
- SpecCheck4.spec
- SpecCheck.spec
- SpecCheckTemp.spec
- unversioned-explicit-obsoletes.spec
- unversioned-explicit-version.spec
- use-of-RPM-SOURCE-DIR.spec
